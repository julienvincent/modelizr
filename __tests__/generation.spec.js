import { models, query, mutate } from './client'
import { PersonWithFriend, PersonWithPets, PersonWithPetsWithFriend } from './client/fragments'

const {Person} = models

test("Query generation for Model with children", () => {
	query(PersonWithFriend).generate(query => {
		expect(query).toMatchSnapshot()
	})
})

test("Query generation for Model with a Union", () => {
	query(PersonWithPets).generate(query => {
		expect(query).toMatchSnapshot()
	})
})

test("Query generation for a modified fragment", () => {
	query(PersonWithPetsWithFriend).generate(query => {
		expect(query).toMatchSnapshot()
	})
})

test("Query generation for query and Model with parameters", () => {
	const model = PersonWithFriend({id: 1, arrayParameter: [1, "2"]})
	const queryParams = {parameter: 1, nestedParameter: {stringParameter: ""}}

	query(queryParams, model).generate(query => {
		expect(query).toMatchSnapshot()
	})

	test("Mutation query generation", () => {
		mutate(queryParams, model).generate(query => {
			expect(query).toMatchSnapshop()
		})
	})
})

test("Query generation for model using .only", () => {
	query(Person.only(["id"])).generate(query => {
		expect(query).toMatchSnapshot()
	})
})

test("Query generation for model using .without", () => {
	query(Person.without(["name"])).generate(query => {
		expect(query).toMatchSnapshot()
	})
})

test("Query generation for model using .empty", () => {
	query(Person.empty()).generate(query => {
		expect(query).toMatchSnapshot()
	})
})