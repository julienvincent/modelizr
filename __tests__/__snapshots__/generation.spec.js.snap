exports[`test Query generation for Model with a Union 1`] = `
"query modelizr_query {
  Person {
    id,
    name,
    age,
    Pets {
      ... on Cat {
        __type,
        id,
        name
      },
      ... on Dog {
        __type,
        ID,
        name,
        breed
      }
    }
  }
}"
`;

exports[`test Query generation for Model with children 1`] = `
"query modelizr_query {
  Person {
    id,
    name,
    age,
    Friend {
      id,
      name,
      age
    }
  }
}"
`;

exports[`test Query generation for a modified fragment 1`] = `
"query modelizr_query {
  Person {
    id,
    name,
    age,
    Pets {
      ... on Cat {
        __type,
        id,
        name
      },
      ... on Dog {
        __type,
        ID,
        name,
        breed
      }
    },
    Friend {
      id,
      name,
      age
    }
  }
}"
`;

exports[`test Query generation for model using .empty 1`] = `
"query modelizr_query {
  Person 
  
}"
`;

exports[`test Query generation for model using .only 1`] = `
"query modelizr_query {
  Person {
    id
  }
}"
`;

exports[`test Query generation for model using .without 1`] = `
"query modelizr_query {
  Person {
    id,
    age
  }
}"
`;

exports[`test Query generation for query and Model with parameters 1`] = `
"query modelizr_query(parameter: 1,nestedParameter: {stringParameter:\"\"}) {
  Person(id: 1,arrayParameter: [1,\"2\"]) {
    id,
    name,
    age,
    Friend {
      id,
      name,
      age
    }
  }
}"
`;
